backend/server/routers/courses.py

```diff
@@ -322,5 +393,56 @@ def coursesUnlockedWhenTaken(userData: UserData, courseToBeTaken: str):
     }
 
 def unlocked_set(courses_state):
-    """fetch the set of unlocked courses from the courses_state of a getAllUnlocked call"""
+    """ Fetch the set of unlocked courses from the courses_state of a getAllUnlocked call """
     return set(course for course in courses_state if courses_state[course]['unlocked'])
+
+
+def fuzzy_match(course: tuple, search_term: str):
+    """ Gives the course a weighting based on the relevance to the search """
+    (code, title) = course
+
+    # either match against a course code, or match many words against the title
+    # (not necessarily in the same order as the title)
+    search_term = search_term.lower()
+    if re.match('[a-z]{4}[0-9]', search_term):
+        return fuzz.ratio(code.lower(), search_term)
+
+    return max(fuzz.ratio(code.lower(), search_term),
+               sum([fuzz.partial_ratio(title.lower(), word)
```

Comment:
You don't need the brackets here. You can pass a generator directly into sum. I usually wouldn't care, but this may be memory intensive if we decide to store the whole list in memory.

Reply:
cool